core/expect.ts

```diff
@@ -228,6 +228,30 @@ export class Matcher {
       }
    }
 
+   public async toThrowAsync(): Promise<void> {
+
+      return new Promise<void>(async (resolve, reject) => {
```

Comment:
I don't believe it's necessary to return a `Promise` in this scenario as you've already marked it as `async` and are calling `await` on the function.